### ISSUE TITLE
Suggested submodule updating in README doesn't update

### ISSUE THREAD

**Issue by ssorallen** [anishathalye/dotbot#13]
`git submodule update --init --recursive` doesn't do what the README seems to be suggesting it for because the setup suggests a bare `git submodule add` for dotbot itself. `submodule add` checks out the repo with a detached HEAD, which makes subsequent updates do nothing because the submodule is not tracking any branch.

To get the desired behavior you could check out the submodule and point it to the master branch:

```
# add submodule to track master branch
git submodule add -b master https://github.com/anishathalye/dotbot

# update your submodule
git submodule update --remote
```

Detailed info: http://stackoverflow.com/a/9189815/368697

**Comment by ssorallen** [comment]
I believe the command could then become:

    git submodule update --init --recursive --remote

**Comment by ssorallen** [comment]
Nix that... if you do both `--recursive` and `--remote` you will point the dotbot submodule to new dependencies that it likely doesn't want. Two commands looks like it will work:

```
- ["git submodule update --init --recursive", "Installing/updating submodules"]                                                                             
- ["git submodule update --remote", "Updating remote tracking submodules"] 
```

**Comment by anishathalye** [comment]
Because of this, I just noticed a typo that was fixed in c32b6f0. Thanks!

I don't actually want to update Dotbot or any other submodules automatically in the `install.conf.yaml`. Fixing it on a specific version (hash) is good for security, and it prevents things from unexpectedly breaking or anything else weird happening.

If someone wants to upgrade Dotbot, they can manually run `git submodule update --remote dotbot`.

As you point out, the README isn't exactly clear on what it's doing. Thanks for pointing that out!

I fixed that in 37846b7. Let me know if you think I should make any other changes.
